slu-status/src/slu-status/slu-status.service.ts

```diff
@@ -0,0 +1,43 @@
+import { BadRequestException, Injectable, Logger } from '@nestjs/common';
+import { SluStatusDto } from './model/SluStatusDto';
+import { SluStatus } from './schema/slu-status.schema';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { Status } from './model/Status';
+
+@Injectable()
+export class SluStatusService {
+	private readonly logger = new Logger(SluStatusService.name);
+
+	constructor(@InjectModel(SluStatus.name) private sluStatusModel: Model<SluStatus>) {}
+
+	async saveSluStatus(sluStatus: SluStatusDto): Promise<SluStatus> {
+		try {
+			const savedSluStatus = (await new this.sluStatusModel(sluStatus).save()).toObject();
+			delete savedSluStatus._id;
+			return savedSluStatus;
+		} catch (ex: any) {
+			this.logger.error(ex.message);
+			throw new BadRequestException(ex.message);
+		}
+	}
+	async updateSluStatus(id: string, status: Status): Promise<SluStatus> {
+		try {
```

Comment:
Please simplify the method: the body should be simply:
```
return await this.sluStatusModel.findByIdAndUpdate(id, { status: status }, { new: true });
```

Reply:
I am using findOneAndUpdate({ id: id }, { status: status }, { new: true, fields: { _id: 0 } }) 
I assume that the user who is requesting this microservice will use the id(did:iota:....)  of the SLU and not the ObjectId of Mongo. Or is the assumption wrong?